src/Guesser/Guess/DateTimeType.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Joli\Jane\Guesser\Guess;
+
+use Joli\Jane\Generator\Context\Context;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Name;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Scalar;
+
+/**
+ * Represent a DateTime type.
+ */
+class DateTimeType extends ObjectType
+{
+    private $format;
+
+    public function __construct($object, $format = DATE_RFC3339)
+    {
+        parent::__construct($object, '\DateTime', []);
```

Comment:
After thinking, i will treat DateTimeImmutable later with a global parameter when generating : `$jane->generateImmutable()`

This generator will create every method of models and generate the correct normalizer code to have a immutable.

I think it make more sense : either having nothing immutable or everything